codegen/templates/augmented_mock.tmpl

```diff
@@ -41,10 +48,13 @@ type {{$methodMockType}} struct {
 {{$methodMockMethod := printf "Expect%s" $methodName -}}
 // {{$methodMockMethod}} returns an object that allows the caller to choose expected scenario for {{$methodName}}
 func (m *MockClientWithFixture) {{$methodMockMethod}}() *{{$methodMockType}} {
-	return &{{$methodMockType}}{
-		scenarios:  m.fixture.{{$methodName}},
-		mockClient: m.MockClient,
+	if m.{{$methodMockField}} == nil {
```

Comment:
do we ever have to concern about # of fixtures we load in memory?

Reply:
We might need to think about it when we have tons of unique fixtures.